services/mock-service/src/fsp-integration/nedbank/nedbank.mock.service.ts

```diff
@@ -0,0 +1,181 @@
+              Path: '',
+              Url: '',
+            },
+          ],
+        };
+      }
+      // Loop over NedbankVoucherStatus enum and check if the orderCreateReference is conttains the enum value
+      for (const status in NedbankVoucherStatus) {
+        if (orderCreateReference.includes(NedbankVoucherStatus[status])) {
+          return {
+            Data: {
+              Transactions: {
+                Voucher: {
+                  Status: NedbankVoucherStatus[status],
+                },
+              },
+            },
+          };
+        }
+      }
+    }
+    return {
+      Data: {
+        Transactions: {
+          Voucher: {
+            Status: NedbankVoucherStatus.REDEEMED,
+          },
+        },
+      },
+    };
+  }
+
+  public async createOrder(
+    createOrderPayload: NedbankCreateOrderMockPayload,
+  ): Promise<Record<string, unknown>> {
+    const { Data } = createOrderPayload;
+    const { Initiation } = Data;
+
+    // This error of max 30 chars is currently not in the integration tests
+    // because it should never happen as paymentReference set using a combination of fields that should never exceed 30 characters
+    // However this if statement is here to catch any potential future changes that might cause this error
+    // So than the happy path test will fail
+    if (
+      Data.Initiation.CreditorAccount.Name.length > 30 ||
+      Data.Initiation.DebtorAccount.Name.length > 30
+    ) {
+      return {
+        Message: 'BUSINESS ERROR',
+        Code: 'NB.APIM.Field.Invalid',
+        Id: '1d3e3076-9e1c-4933-aa7f-69290941ec70',
+        Errors: [
+          {
+            ErrorCode: 'NB.APIM.Field.Invalid',
+            Message: 'Request Validation Error - Name is too long',
+          },
+        ],
+      };
+    }
+
+    // Scenario  Incorrect DebtorAccount Identification
+    if (
+      !Initiation.DebtorAccount.Identification ||
+      Data.Initiation.CreditorAccount.Identification.includes(
+        NedbankMockNumber.failDebitorAccountIncorrect,
+      )
+    ) {
+      return {
+        Message: 'BUSINESS ERROR',
+        Code: 'NB.APIM.Field.Invalid',
+        Id: '1d3e3076-9e1c-4933-aa7f-69290941ec70',
+        Errors: [
+          {
+            ErrorCode: 'NB.APIM.Field.Invalid',
+            Message:
+              'Request Validation Error - TPP account configuration mismatch',
+            Path: '',
+            Url: '',
+          },
+        ],
+      };
+    }
+    if (
+      Data.Initiation.CreditorAccount.Identification.includes(
+        NedbankMockNumber.failTimoutSimulate,
+      )
+    ) {
+      // Simulate a timeout error
+      // This is not an actual timeout since testing an actual timeout would make our automated test runs slower
+      // This is just a way to simulate the result of a timout error which would be that the order status would never be updated
+      // ##TODO: discuss if we can find a more elegant way to simulate a timeout error
```

Comment:
Looks fine to me. What is not clean about this solution?

Reply:
There is not an actual timeout happening maybe something unexpected happens here. Maybe something unexpected happens if we change the way we handle timeouts in the 121-service in general. But I think it's hard to simulate this precisely and it's also a relevant scenario for all other fsps, I guess it's easier to just leave it like this.